module/assets/helper.js

```diff
@@ -0,0 +1,1075 @@
+                       code: 'PROGRAM_MANAGER',
+                       programs: [
+                           607d320de9cce45e22ce90c0,
+                                ]
+                      },
+                  ]
+               */
+                let platformRolesToUpdate =
+                  await this.fetchPlatformRolesToTransfer(fromUserData);
+
+                //create user if not exits in the user extension collection
+                let createUserExtension =
+                  await userExtensionsHelper.createOrUpdate(
+                    [],
+                    addUserExtension
+                  );
+                if (createUserExtension) {
+                  let matchQuery = {
+                    userId: reqData.toUserProfile.userId,
+                  };
+
+                  let pushQuery = {
+                    $push: {
+                      platformRoles: {
+                        $each: platformRolesToUpdate,
+                      },
+                    },
+                  };
+
+                  let fromUserMatchQuery = {
+                    userId: reqData.fromUserProfile.userId,
+                  };
+                  let setQuery = {
+                    $set: { platformRoles: [] },
+                  };
+                  let updateUserExtension =
+                    await userExtensionsHelper.updateMany(
+                      matchQuery,
+                      pushQuery
+                    );
+                  if (updateUserExtension) {
+                    await userExtensionsHelper.updateMany(
+                      fromUserMatchQuery,
+                      setQuery
+                    );
+                  }
+                }
+              }
+              //Queries to update owner in programs Collections for ownership transfer
+              let programFilter = {
+                owner: reqData.fromUserProfile.userId,
+              };
+              let updatePrograms = {
+                $set: {
+                  owner: reqData.toUserProfile.userId,
+                },
+              };
+              let updatePartialPrograms = await programsHelper.updateMany(
+                programFilter,
+                updatePrograms
+              );
+              updatePartialPrograms
+                ? (updateUserAssetDataResult = true)
+                : (updateUserAssetDataResult = false);
+            }
+          } else {
+            //condition for if there is an assetInformation in ownershipTransferEvent
+
+            let typeOfAssetsToMove = reqData.assetInformation.objectType;
+            if (
+              reqData.toUserProfile.roles.includes(
+                constants.common.CONTENT_CREATOR
+              ) &&
+              typeOfAssetsToMove === constants.common.SOULTION
+            ) {
+              //match and Set Queries for solutions partial Transfer
+              let solutionFilter = {
+                author: reqData.fromUserProfile.userId,
```

Comment:
@praveenKDass _id filter should be first and then author.Refer - https://www.mongodb.com/docs/manual/tutorial/equality-sort-range-rule/